.github/workflows/ci.yml

```diff
@@ -335,169 +350,125 @@ jobs:
             **/surefire-reports/TEST-*.xml
           retention-days: ${{ env.TEST_REPORT_RETENTION_DAYS }}
 
-  test:
+  build-test-matrix:
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        modules:
-          - ":trino-main"
-          - ":trino-tests"
-          - ":trino-raptor-legacy"
-          - ":trino-accumulo"
-          - ":trino-cassandra"
-          - ":trino-clickhouse"
-          - ":trino-hive"
-          - ":trino-hive -P test-parquet"
-          - ":trino-hive -P test-failure-recovery"
-          - ":trino-hive -P test-fault-tolerant-execution"
-          - ":trino-elasticsearch,:trino-elasticsearch -P test-stats"
-          - ":trino-mongodb"
-          - ":trino-kafka"
-          - ":trino-pinot"
-          - ":trino-redis"
-          - ":trino-mysql"
-          - ":trino-postgresql"
-          - ":trino-sqlserver"
-          - ":trino-oracle"
-          - ":trino-kudu"
-          - ":trino-druid"
-          - ":trino-iceberg"
-          - ":trino-phoenix,:trino-phoenix5"
-          - ":trino-jdbc,:trino-base-jdbc,:trino-thrift,:trino-memory"
-    timeout-minutes: 60
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # checkout all commits to be able to determine merge base for GIB
+      - name: Fetch base ref to find merge-base for GIB
+        run: .github/bin/git-fetch-base-ref.sh
       - uses: actions/setup-java@v2
         with:
           distribution: 'zulu'
           java-version: 11
           cache: 'maven'
```

Comment:
This should be removed from this job. It takes 3 minutes with cache, but cache here is not used at all.